developer/migrations.md

```diff
@@ -0,0 +1,66 @@
+# Database migrations
+
+## How we deal with database changes during project life cycle
+Software that evolves will sooner or later have to change existing data in their database in order to be compatible with the latest code changes. Reaction uses the meteor package [percolate:migrations](https://atmospherejs.com/percolate/migrations) for internal database migrations.
+
+### Migrating on startup
+On application startup, Reaction will migrate to the latest database revision. It does so by sourcing the revision files in directory `/imports/plugins/core/versions/server/migrations/*.js` and apply their content's database instructions, if necessary.
+
+**/imports/plugins/core/versions/server/startup.js**
+```
+Hooks.Events.add("afterCoreInit", () => {
+  Migrations.migrateTo("latest");
+});
+```
+
+The Migrations package keeps track of the applied changes in a dedicated collection aptly called _Migrations_ (configurable):
+![Screenshot of Collection Migrations](../assets/screenshot-migrations-collection.png)
+
+Hint: In case there's an error during a migration run, it could happen that the migration process doesn't release the lock state. If that happens, check for any errors in the up() and down() methods. If they're fixed, you may set the `locked` flag in the database to `false` and try it again.
```

Comment:
In case there's an error during a migration run, it is possible that the migration process didn't release the lock state.

Reply:
fixed.